.github/gha_monitor/gha_failures.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+import logging
+import json
+import os
+
+from modules import github
+from modules import sns
+from datetime import datetime, timedelta
+from dateutil import parser, tz
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+# NOTE: this should roughly match the times for cron in the workflow yaml.
+# e.g. if cron is set to run every hour, give some wiggle room and use hours=1.1
+TIME_WINDOW_BEGIN = datetime.utcnow().astimezone(tz.UTC) - timedelta(hours=1.1)
+TIME_WINDOW_END = datetime.utcnow().astimezone(tz.UTC)
+ONCALL_WIKI = "https://w.amazon.com/bin/view/AWSCryptography/AWSCryptoTools/TransportLibraries/OnCall/"
+
+
+class GitHub_Actions(github.GitHub_Client):
+    params = {
+        # Needed when using an API key
+        'github_username': os.getenv('github_username'),
+        'github_password': os.getenv('github_password'),
+        # Use from within an Action - ignored if username set
+        'token': os.getenv('GITHUB_TOKEN'),
+        'repo_organization': os.getenv('GITHUB_REPO_ORG'),
+        'repo': os.getenv('GITHUB_REPO')
+    }
+
+
+class S2n_Notices(sns.SNS_Client):
+    params = {
+        'topic_arn': 'arn:aws:sns:us-west-2:024603541914:s2n_notices'
+    }
+
+
+def workflow_filter(log):
+    """ Reduce the number of elements from GitHub for JSON SNS. """
+    top_level_keys = set(['head_branch', 'head_commit', 'status', 'conclusion', 'html_url', 'created_at', 'repository'])
+    to_drop = set(log) - top_level_keys
+    for element in to_drop:
+        del log[element]
+    logging.debug('Filtered json: {log}')
+    return log
+
+
+def message_text():
+    """ Formatting for text message. """
+    return """
+s2n GitHub Action monitor notice
+State: failed
```

Comment:
Along with the failed status, can we print out the cause of failure? Am error message or error code?

Reply:
It's not captured by the action, this would require downloading a log bundle, unzipping and using grep to guess at the cause- in this example case, the action never even started, so there is no error.

https://developer.github.com/v3/actions/workflow_runs/#list-workflow-run-logs